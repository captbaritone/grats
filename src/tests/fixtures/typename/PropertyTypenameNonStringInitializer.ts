/** @gqlType */
export class User {
  __typename = 1 as const;
  /** @gqlField */
  name: string = "Alice";
}
