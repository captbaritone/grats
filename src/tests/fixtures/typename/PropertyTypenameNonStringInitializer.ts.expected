-----------------
INPUT
----------------- 
/** @gqlType */
export class User {
  __typename = 1 as const;
  /** @gqlField */
  name: string = "Alice";
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/typename/PropertyTypenameNonStringInitializer.ts:3:16 - error: Expected `__typename` property initializer to be a string literal. For example: `__typename = "MyType"` or `__typename: "MyType";`. This lets Grats know that the GraphQL executor will be able to derive the type of the object at runtime.

3   __typename = 1 as const;
                 ~
